recipes/demjson3/0001-Add-entry-points.patch

```diff
@@ -0,0 +1,88 @@
+From 07a2a2270e782de89a0226c3f93fdb674aff5ab9 Mon Sep 17 00:00:00 2001
+From: Marco Bonifacio <bonifacio.marco@gmail.com>
+Date: Wed, 23 Feb 2022 12:32:12 +0100
+Subject: [PATCH] Add entry points
+
+---
+ jsonlint.py | 50 ++++++++++++++++++++++++++++++++++++++++++++++++++
+ setup.py    |  7 +++++--
+ 2 files changed, 55 insertions(+), 2 deletions(-)
+ create mode 100644 jsonlint.py
+
+diff --git a/jsonlint.py b/jsonlint.py
+new file mode 100644
+index 0000000..c674f1f
+--- /dev/null
++++ b/jsonlint.py
+@@ -0,0 +1,50 @@
++#!/usr/bin/env python
++# -*- coding: utf-8 -*-
++r"""A JSON syntax validator and formatter tool.
++
++Requires demjson module.
++
++"""
++__author__ = "Deron Meranda <http://deron.meranda.us/>"
++__homepage__ = "http://deron.meranda.us/python/demjson/"
++__date__ = "2014-12-22"
++__version__ = "2.2.4"
++__credits__ = """Copyright (c) 2006-2015 Deron E. Meranda <http://deron.meranda.us/>
++
++Licensed under GNU LGPL (GNU Lesser General Public License) version 3.0
++or later.  See LICENSE.txt included with this software.
++
++This program is free software: you can redistribute it and/or modify
++it under the terms of the GNU Lesser General Public License as
++published by the Free Software Foundation, either version 3 of the
++License, or (at your option) any later version.
++
++This program is distributed in the hope that it will be useful,
++but WITHOUT ANY WARRANTY; without even the implied warranty of
++MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
++GNU General Public License for more details.
++
++You should have received a copy of the GNU Lesser General Public License
++along with this program.  If not, see <http://www.gnu.org/licenses/>
++or <http://www.fsf.org/licensing/>.
++
++"""
++
++import sys
++
++try:
++    import demjson3
++except ImportError:
++    sys.stderr.write("Can not import the demjson Python module.\n")
++    sys.stderr.write("Try running:  pip install demjson\n")
++    sys.exit(1)
++
++
++def main():
++    lint = demjson3.jsonlint(program_name=sys.argv[0])
++    rc = lint.main( sys.argv[1:] )
++    sys.exit(rc)
++
++
++if __name__ == '__main__':
++    main()
```

Comment:
Is all of the stuff above this really required? It should just be a change in entry points, I would assume.

Reply:
Perhaps the name of the patch isn't clear enough, in addition to the entry point the original script has to be converted in a py module with a proper main function to hook the entry point. I tested locally the patch and the CLI script correctly works on Windows now.